orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/EvaluateConditionTask.java

```diff
@@ -0,0 +1,113 @@
+
+@Component
+public class EvaluateConditionTask implements RetryableTask {
+  private static Logger log = LoggerFactory.getLogger(EvaluateConditionTask.class);
+  private final ConditionConfigurationProperties conditionsConfigurationProperties;
+  private final Optional<List<ConditionSupplier>> suppliers;
+  private final Clock clock;
+
+  @Autowired
+  public EvaluateConditionTask(
+    ConditionConfigurationProperties conditionsConfigurationProperties,
+    Optional<List<ConditionSupplier>> suppliers,
+    Clock clock
+  ) {
+    this.conditionsConfigurationProperties = conditionsConfigurationProperties;
+    this.suppliers = suppliers;
+    this.clock = clock;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return 3000L;
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.SECONDS.toMillis(conditionsConfigurationProperties.getWaitTimeoutSeconds());
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!conditionsConfigurationProperties.isEnabled() || !suppliers.isPresent()) {
+      log.warn("Conditions not configured. Skipping wait.");
+      stage.getContext().put("status", Status.SKIPPED);
+      return new TaskResult(ExecutionStatus.SUCCEEDED, stage.getContext());
+    }
+
+    final WaitForConditionContext ctx = stage.mapTo(WaitForConditionContext.class);
+    if (ctx.getStatus() == Status.SKIPPED) {
+      log.info("Skipping wait. Status: {}", ctx.getStatus());
+      return new TaskResult(ExecutionStatus.SUCCEEDED, stage.getContext(), stage.getOutputs());
+    }
+
+    Instant now = clock.instant();
+    Duration backoff = Duration.ofSeconds(conditionsConfigurationProperties.getBackoffWaitSeconds());
+    Map<String, Object> tmp = new HashMap<>();
+    tmp.put("startTime", Optional.ofNullable(ctx.getStartTime()).orElse(now));
+    if (ctx.getStartTime() != null && ctx.getStartTime().plus(backoff).isAfter(now)) {
+      tmp.put("status", Optional.ofNullable(ctx.getStatus()).orElse(Status.WAITING));
+      return new TaskResult(ExecutionStatus.RUNNING, tmp, stage.getOutputs());
+    }
+
+    try {
+      Set<Condition> conditions = suppliers.get()
+        .stream()
+        .flatMap(supplier -> supplier.getConditions(stage).stream()).filter(Objects::nonNull)
+        .collect(Collectors.toSet());
+
+      final Status status = conditions.isEmpty() ? Status.SKIPPED : Status.WAITING;
+      tmp.put("status", status);
+      stage.getOutputs().put("conditions", conditions);
+      if (status == Status.WAITING) {
+        return new TaskResult(ExecutionStatus.RUNNING, tmp, stage.getOutputs());
+      }
+
+      return new TaskResult(ExecutionStatus.SUCCEEDED, tmp, stage.getOutputs());
+    } catch (Exception e) {
+      log.error("Error occurred while fetching for conditions to eval.", e);
+      throw new IllegalStateException("Failure to retrieve conditions for evaluation.", e);
```

Comment:
Unclear if we want an error in a conditions stage to fail the pipeline?

Reply:
What should the appropriate behavior be in your opinion? Should the pipeline not fail if for some reason a failure occurs while checking conditions?

Reply:
To start, I'd probably err on the side of it not blowing up the pipeline.